apps/sms/views/conversation/test/unit/conversation_test.js

```diff
@@ -2887,13 +2889,18 @@ suite('conversation.js >', function() {
         assert.equal(button.dataset.l10nId, 'download-attachment');
       });
       suite('clicking', function() {
-        var showMessageErrorSpy;
+        var showMessageErrorSpy, retrievePromise, onResolve, onReject;
 
         setup(function() {
           if (!('mozSettings' in navigator)) {
            navigator.mozSettings = null;
           }
 
+          retrievePromise = new Promise((resolve, reject) => {
+            onResolve = resolve;
+            onReject = reject;
+          });
+          MessagingClient.retrieveMMS.returns(retrievePromise);
```

Comment:
interesting idea; I never thought of doing it like this (usually I prefer to run `MessagingClient.retrieveMMS.returns(...)` in the test itself so that I can change the return value -- but then I'd need to call handleMessageClick in the test itself as well.

So this is an interesting idea here :)

I don't really like the names `onResolve` and `onReject`; maybe `doResolve` and `doReject` feel better. Also can you nullify these variable in a `teardown` function ? Just a cleanup :)